app/controllers/rooms.js

```diff
@@ -196,12 +204,42 @@ module.exports = function() {
                     return res.sendStatus(400);
                 }
 
-                var user = req.user.toJSON();
-                user.room = room._id;
+                var password = req.data.password;
+                if(!!room.password && !password) {

```

Comment:
@funkaoshi I usually only do it, if I don't want the original value to be leaked.

For example, the following method can be used to test if the object has a password. A [non-empty string](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) will evaluate to true.

```
function hasPassword() {
  return this.password;
}
```

But it leaks the password value - so to prevent this, you can use ```!!```:

```
function hasPassword() {
  return !!this.password;
}
```